### PR TITLE
feature: filter for raw query strings

### PR DESCRIPTION
**Pull request by hwchen**
A new filter `raw_query()` creates a `Filter` that returns the raw query
string as type String.

- new fn `filters::query::raw_query()`
- expose fn publicly as `warp::raw_query()` in lib.rs
- simple test in `tests/query/rs`

### CODE REVIEW

**Comment by hwchen** [comment]
This would resolve https://github.com/seanmonstar/warp/issues/22 by allowing custom query string parsing.

File: src/lib.rs [hunk 1]
```diff
@@ -130,6 +130,8 @@ pub use self::filters::{
     query,
     // query() function
     query::query,
+    // raw_query() function
+    query::raw_query,
```
Comment (seanmonstar) [inline src/lib.rs]: This is just a thought, but what do you think if the function were just `raw`, and not re-exported, just left in the `query` module? That'd mean typical usage would then be `warp::query::raw()`...

Comment (hwchen) [reply src/lib.rs]: Makes sense to me, this way it won't seem like a top-level filter. I'll make the change.

**Comment by seanmonstar** [comment]
Great, thanks!

**Comment by hwchen** [comment]
No problem! Thanks!

**Pull request merged**